probe/endpoint/connection_tracker.go

```diff
@@ -0,0 +1,258 @@
+package endpoint
+
+import (
+	"strconv"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/weaveworks/scope/probe/endpoint/procspy"
+	"github.com/weaveworks/scope/probe/process"
+	"github.com/weaveworks/scope/report"
+)
+
+// connectionTrackerConfig are the config options for the endpoint tracker.
+type connectionTrackerConfig struct {
+	HostID       string
+	HostName     string
+	SpyProcs     bool
+	UseConntrack bool
```

Comment:
`UseConntrack` determine if we use conntrack or not independently from the other way of detecting connections. While `WalkProc` determine if we use proc parsing or not.
I designed `connectionTracker` in a general way, so if ebpf is enabled then the internal logic will decide to not use proc parsing and/or conntrack. However, the configuration comes from the user that set the conntrack flag.